exporter/collector/internal/integrationtest/metric_test_server.go

```diff
@@ -151,19 +150,19 @@ func (m *MetricsTestServer) NewExporter(
 	ctx context.Context,
 	t testing.TB,
 	cfg collector.Config,
-) component.MetricsExporter {
-	factory := collector.NewFactory()
+) *collector.MetricsExporter {
 	cfg.Endpoint = m.Endpoint
 	cfg.UseInsecure = true
 	cfg.ProjectID = "fakeprojectid"
 
-	exporter, err := factory.CreateMetricsExporter(
+	exporter, err := collector.NewGoogleCloudMetricsExporter(
```

Comment:
> Remove tests for exporterhelper-produced metrics. These will live upstream.

I'm a little confused how these would look upstream, would we move the whole integration testing setup there?

IIRC you were going to duplicate the factory + exporterhelper code in a separate integration test module so that the tests are as realistic as possible.

Reply:
It wouldn't be full end-to-end tests if they are upstream.  It would have to use something like a mock to check and see if metrics are recorded and incremented correctly.  Alternatively, the unit testing in the exporterhelper library itself may be sufficient.  Maybe we just need to check to see if the metrics are registered or something else super-simple.

Reply:
updated the PR description to capture this.